### ISSUE TITLE
Swipeable with animated `renderRightAction` working weird on Android

### ISSUE THREAD

**Issue by SpertsyanKM** [software-mansion/react-native-gesture-handler#2471]
### Description

I have a Swipeable component which renders right actions on swipe and is removed if released after a threshold. Everything works correctly on iOS, but on Android is does not animate appearing, but animates returning back. Here are videos of how it works on both OSs:

https://user-images.githubusercontent.com/13959241/236048330-30906f8d-4051-492f-943a-eb2bad2663cd.mov

https://user-images.githubusercontent.com/13959241/236048357-d02af3f9-ba21-4a62-9154-c5dbbb826aaa.mp4

### Steps to reproduce

My component:

```
  const renderRightActions = (
    progress: Animated.AnimatedInterpolation<number>,
    dragX: Animated.AnimatedInterpolation<number>,
  ) => {
    const trans = dragX.interpolate({
      inputRange: [-500, 0],
      outputRange: [SCREEN_WIDTH - 500, SCREEN_WIDTH],
    });
    return (
      <TransactionRemoveButton
        style={{
          transform: [{translateX: trans}],
        }}
      />
    );
  };

  const onSwipeableOpen = useCallback(() => {
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    dispatch(deleteTransaction(transaction));
  }, [dispatch, transaction]);

  return (
    <Swipeable
      renderRightActions={renderRightActions}
      onSwipeableOpen={onSwipeableOpen}>
      <TouchableRipple onPress={() => onPress(transaction)}> // from react-native-pape
        <>
          ...
        </>
      </Container>
    </Swipeable>
  );
```

TransactionRemoveButton, used in `renderRightActions` (uses styled-components):

```
export const Container = styled(Animated.View)`
  background-color: ${Color.Red};
  flex: 1;
`;

export const Content = styled(View)`
  padding-horizontal: 30px;
  height: 100%;
  align-items: flex-start;
  justify-content: center;
`;

type Props = {
  style?: StyleProp<ViewStyle>;
};

export const TransactionRemoveButton: React.FC<Props> = ({style}) => {
  return (
    <Container style={style}>
      <Content>
        <TrashBinIcon color={Color.White} /> // is just an SVG icon
      </Content>
    </Container>
  );
};

```

### Snack or a link to a repository

https://snack.expo.dev/@spertsyankm/bold-beef-jerky?platform=android

### Gesture Handler version

2.8.0

### React Native version

0.71.7

### Platforms

Android

### JavaScript runtime

None

### Workflow

React Native (without Expo)

### Architecture

None

### Build type

None

### Device

Android emulator

### Device model

Pixel 4 API 30 (Android 11)

### Acknowledgements

Yes

**Comment by SpertsyanKM** [comment]
Added: https://snack.expo.dev/@spertsyankm/bold-beef-jerky?platform=android.

**Comment by SpertsyanKM** [comment]
But anyway my problem was a bit different. In the snack swibe gesture wasn't working at all, while in my example it works only after release (the first video of the original message). Maybe that's the reason the snack began working, while my app didn't.

**Comment by SpertsyanKM** [comment]
I don't know, what is the reason. I replaced my complete app with the snack code and it behaves as on the video. The only difference between snack and my app is that I use pure react native and snack uses expo. Can it be the problem?

The version of the react-native-gesture-handler is the same..

**Comment by SpertsyanKM** [comment]
Well, I've updated my dependency version from 2.8.0 to 2.9.0 and the problem has been resolved. 

By the way, first I tried version 2.10.0, where this problem is also resolved, but there became another problem - `TouchableRipple` (from react-native-paper) on iOS was receiving click-events even when I was just swiping a swipeable container it was placed in. So I've downgraded to 2.9.0 where both problems don't exist.
